CONTRIBUTING.md

```diff
@@ -15,12 +15,63 @@ us first through the issue tracker with your idea so that we can help out and
 possibly guide you. Coordinating up front makes it much easier to avoid
 frustration later on.
 
-### Code reviews
+## Code reviews
+
 All submissions, including submissions by project members, require review. We
 use Github pull requests for this purpose.
 
-### The small print
+The submitted code should adhere to the following:
+
+### Do's
+
+* For every pull request, make sure all commits are consolidated into one commit before
+  sending the pull request.
+* Before submitting any code changes, please ensure that the [Unit Tests](https://github.com/google/EarlGrey/tree/master/Tests/UnitTests)
+  and [Functional Tests](https://github.com/google/EarlGrey/tree/master/Tests/FunctionalTests)
+  projects build and all test pass. If you believe that the test failures are due to bugs in the
+  framework then please [file an issue](https://github.com/google/EarlGrey/issues).
+* If a pull request looks good then we'll wait for Travis to run all the Unit and
+  Functional tests on the changes. We also trigger an internal build to check for
+  further build breakages. In case any of the tests are broken, we will let you know.
+* If your pull request does not touch any code, then you could skip the Travis Continuous
+  Integration check by adding [ci skip](https://docs.travis-ci.com/user/customizing-the-build/#Skipping-a-build)
```

Comment:
[[ci skip]](https://docs.travis-ci.com/user/customizing-the-build/#Skipping-a-build)